### PR TITLE
Fixes: kOSToolbarWindow, bootFile can be set to None in VAB/SPH

### PR DESCRIPTION
**Pull request by ZiwKerman**
#761 

Fixes some kOSToolbarWindow long standing issues. Rewrote texture handling to use GameDatabase loaded textures and made textures static, so there are no multiple instances of textures.

BTW @Dunbaratu - I removed Instance counting as it is really unnecessary anymore, every instance created gets destroyed properly (left some logging in Start and OnDestroy).

### CODE REVIEW

**Comment by ZiwKerman** [comment]
also fixes #763

**Comment by Dunbaratu** [comment]
This looks like a good change.  I wrote that kostoolbar stuff based entirely on a post in the forums by maxmaps that made very little sense to me, and had to add all the instance counting because it kept throwing exceptions when doing it the way his post said to - frustrating sensitive timing exceptions that you can't trace without knowing a lot about Unity itself, which I don't.

I've long known it was bad, but had no idea what I was meant to do instead because the only documentation was that one post that didn't explain it well.

Thanks for the fix.  I'll be testing it out today.

**Comment by ZiwKerman** [comment]
One of the major things I added is this 2 lines in OnDestroy (GoAway)

```
   GameEvents.onGUIApplicationLauncherReady.Remove(RunWhenReady);
   GameEvents.onGUIApplicationLauncherDestroyed.Remove(GoAway);
```
Because they were missing the instances of kOSToolbarWindow did not despawn properly, but now all should be fine

**Comment by Dunbaratu** [comment]
Yeah the original post by maxmaps never mentioned making calls to GameEvents, or what it was.  It just said to set up things this way and it will work... which it didn't.

**Comment by Dunbaratu** [comment]
I will be testing the toolbar stuff.  I am in no position to test the bootfile stuff that is also intermingled into this PR, as I know nothing of that system, and can't compare what I see to how it worked before.

File: src/kOS/Screen/KOSToolBarWindow.cs [hunk 1]
```diff
@@ -80,100 +80,65 @@ public class KOSToolBarWindow : MonoBehaviour
...
-            terminalOpenTelnetIconImage.LoadImageIntoTexture(terminalOpenTelnetIconTexture);
-            terminalClosedTelnetIconImage.LoadImageIntoTexture(terminalClosedTelnetIconTexture);
-
-            windowRect = new Rect(0,0,width,height); // this origin point will move when opened/closed.
+            launcherButtonTexture = GameDatabase.Instance.GetTexture ("kOS/GFX/launcher-button",false);
+            terminalOpenIconTexture = GameDatabase.Instance.GetTexture ("kOS/GFX/terminal-icon-open", false);
+            terminalClosedIconTexture = GameDatabase.Instance.GetTexture ("kOS/GFX/terminal-icon-closed", false);
+            terminalOpenTelnetIconTexture = GameDatabase.Instance.GetTexture ("kOS/GFX/terminal-icon-open-telnet", false);
+            terminalClosedTelnetIconTexture = GameDatabase.Instance.GetTexture ("kOS/GFX/terminal-icon-closed-telnet", false);
+
```
Comment (Dunbaratu) [inline src/kOS/Screen/KOSToolBarWindow.cs]: At first I didn't like this change away from using constant strings for the names, but now that I think of it, there's nowhere else we'd be using those strings anyway except for right here, so I guess this is okay.

**Comment by Dunbaratu** [comment]
The KOSToolbarWindow portion of this PR has my blessing.  I read the code changes and tested the following cases:

  * revert to launch with kos panel open.
  * revert to launch with kos panel closed.
  * revert to VAB, then relaunch, with kos panel open.
  * revert to VAB, then relaunch, with kos panel closed.
  * new craft in VAB, with kos panel open in VAB, then launch.
  * new craft in VAB, with kos panel closed in VAB, then launch.

The strange behaviors do indeed seem to be gone now.

However, I have not tested (and will not be testing) the associated boot file change that is also in the same PR.  Can someone else test that part and if it's good, merge this?

File: src/packages/repositories.config [hunk 2]
```diff
@@ -1,4 +1,5 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<repositories>
-  <repository path="..\kOS.Safe.Test\packages.config" />
```
Comment (erendrake) [inline src/packages/repositories.config]: what OS/filesystem are you using for development?

Comment (ZiwKerman) [inline src/packages/repositories.config]: this is maybe from my xamarin/monodevelop

**Comment by erendrake** [comment]
@ZiwKerman Thank you :)

In the future i would prefer to have 1 PR per issue unless multiple issues are very closely related. that way we can test the one change and merge it.

**Comment by ZiwKerman** [comment]
Mea culpa

The changes seemed so small at first :)

**Pull request merged**